openage/convert/dataformat/format_metadata.py

```diff
@@ -0,0 +1,154 @@
+class DataDefinition:
+    version = None
+    type = None
+
+    def __init__(self, data, name):
+        self.data = data
+        self.name = name
+        self.txt = []
+
+    def add_header(self):
+        self.txt.append('# openage {} definition for {}\n'.format(self.type, self.name))
+
+    def add_version(self):
+        self.txt.append('version {}\n'.format(self.version))
```

Comment:
`self` falls back to the class attribute if the instance does not have that attribute. Is it bad practice to use `self` here?

The reason I used `self` is that this way I can define the function in the superclass `DataDefinition` despite the fact that its `version` and `type` are not defined. Otherwise, I would have to have the header formatting function copy-pasted in the two subclasses.

Reply:
I would consider it bad practice here because `version` is supposed to be a constant that is fix for all classes. It should not be altered by instances. By accessing the class attribute, this is more clear and explicit. When code is worked on by multiple people, it is better to aim for clarity than for fewer lines.

`add_header` is not that necessary anyway, since the header is mandatory and could be added in `dump()` later.